Describe block breaks coverage

Any code only executed by a test function inside a describe block is not reported by coverage:

```
def foo():
    return False


def describe_foo():
    def test_bar():
        assert foo()
```

produces:

```
 » vex tmp py.test --cov /tmp/bug.py --cov-report=term-missing /tmp/bug.py                                                               1 ↵
=============================================================================================== test session starts ================================================================================================
platform linux2 -- Python 2.7.6 -- py-1.4.25 -- pytest-2.6.3
plugins: cov, describe
collected 1 items 

../../../../tmp/bug.py F

===================================================================================================== FAILURES =====================================================================================================
______________________________________________________________________________________________ describe_foo.test_bar _______________________________________________________________________________________________

    def test_bar():
>       assert foo()
E       assert foo()

/tmp/bug.py:7: AssertionError
--------------------------------------------------------------------------------- coverage: platform linux2, python 2.7.6-final-0 ----------------------------------------------------------------------------------
Name       Stmts   Miss  Cover   Missing
----------------------------------------
/tmp/bug       5      3    40%   2, 6-7
============================================================================================= 1 failed in 0.01 seconds =============================================================================================
```

--- a/pytest_describe/plugin.py
+++ b/pytest_describe/plugin.py
@@ -7,20 +7,21 @@ from _pytest.python import PyCollector
 def trace_function(funcobj, *args, **kwargs):
     """Call a function, and return its locals"""
     funclocals = {}
+    trigger = {}
 
     def _tracefunc(frame, event, arg):
         if event == 'call':
             # Activate local trace for first call only
             if frame.f_back.f_locals.get('_tracefunc') == _tracefunc:
-                return _tracefunc
-        elif event == 'return':
-            funclocals.update(frame.f_locals)
+                trigger.update({'exit': frame})
+        elif event == 'return' and frame == trigger.get('exit'):
+            funclocals.update(frame.f_locals.copy())
 
-    sys.settrace(_tracefunc)
+    sys.setprofile(_tracefunc)
     try:
         funcobj(*args, **kwargs)
     finally:
-        sys.settrace(None)
+        sys.setprofile(None)
 
     return funclocals
 

